pallets/parachain-staking/src/lib.rs

```diff
@@ -2445,25 +3021,25 @@ pub mod pallet {
 				round_issuance.ideal
 			}
 		}
-		fn delegator_leaves_collator(
+		/// Remove delegation from candidate state
+		/// Amount input should be retrieved from delegator and it informs the storage lookups
+		fn delegator_leaves_candidate(
+			candidate: T::AccountId,
 			delegator: T::AccountId,
-			collator: T::AccountId,
+			amount: BalanceOf<T>,
 		) -> DispatchResult {
-			let mut state = <CandidateState<T>>::get(&collator).ok_or(Error::<T>::CandidateDNE)?;
-			let (total_changed, delegator_stake) = state.rm_delegator::<T>(delegator.clone())?;
-			T::Currency::unreserve(&delegator, delegator_stake);
+			let mut state = <CandidateInfo<T>>::get(&candidate).ok_or(Error::<T>::CandidateDNE)?;
+			let total_changed = state.rm_delegation::<T>(&candidate, delegator.clone(), amount)?;
```

Comment:
From what I am reading, unreserve moves up to `amount`:

https://github.com/paritytech/substrate/blob/ded44948e2d5a398abcb4e342b0513cb690961bb/frame/support/src/traits/tokens/currency/reservable.rs#L60

Can I execute a delegator_leaves_candidate request with an amount higher than what I have reserved? and since we are not checking what is returned from unreserve, would we update the total_changed by the amount I specified or by the amount I have reserved?

Reply:
Ok looking deeper in the pallet I think there is no way a user can unreserve more than what it has bonded before.

Reply:
I think we should still take into account of the return value. It makes the assumption that the rest of the code make this case impossible, which we might forget about and introduce later a new case where this is not checked.

Reply:
I agree

Reply:
I agree, to at least emit some warning message if the assumptions built into the pallet are ever violated. I'll make a jira ticket for this as a follow up, there's a few places it is needed.